packages/big-design/src/components/TableNext/hooks/useSelectable/helpers.ts

```diff
@@ -0,0 +1,209 @@
+  parentRowIndex,
+}: SelectRowArg<T>) {
+  const newSelectedItems = { ...selectedItems };
+
+  delete newSelectedItems[parentRowIndex];
+
+  for (let index = 0; index < childrenRows.length; index++) {
+    const childRowIndex = index;
+
+    delete newSelectedItems[`${parentRowIndex}.${childRowIndex}`];
+  }
+
+  return newSelectedItems;
+}
+
+function selectParent<T>({
+  childrenRows,
+  isExpandable,
+  parentRowIndex,
+  selectedItems,
+}: SelectRowArg<T>): TableSelectable['selectedItems'] {
+  const hasChildrenRows = isExpandable && childrenRows !== undefined;
+
+  // If parent has children, select it's childrenRows
+  if (hasChildrenRows) {
+    const newSelectedItems = selectParentAndChildren({
+      selectedItems,
+      childrenRows,
+      parentRowIndex,
+    });
+
+    return newSelectedItems;
+  }
+
+  return {
+    ...selectedItems,
+    [parentRowIndex]: true,
+  };
+}
+
+function selectParentAndChildren<T>({
+  selectedItems,
+  childrenRows = [],
+  parentRowIndex,
+}: SelectRowArg<T>) {
+  const newSelectedItems = { ...selectedItems };
+
+  newSelectedItems[parentRowIndex] = true;
+
+  for (let index = 0; index < childrenRows.length; index++) {
+    const childRowIndex = index;
+
+    newSelectedItems[`${parentRowIndex}.${childRowIndex}`] = true;
+  }
+
+  return newSelectedItems;
+}
+
+export function selectChildRow<T>({
+  childRowIndex,
+  isTheOnlySelectedChildRow,
+  selectedItems,
+  parentRowIndex,
+}: SelectRowArg<T>): TableSelectable['selectedItems'] {
+  const isSelectedParent = selectedItems[parentRowIndex] !== undefined;
+
+  if (!isSelectedParent) {
+    const newSelectedItems = selectChild({ childRowIndex, parentRowIndex, selectedItems });
+
+    return newSelectedItems;
+  }
+
+  const isSelectedChild = selectedItems[`${parentRowIndex}.${childRowIndex}`] !== undefined;
+
+  if (isSelectedChild) {
+    const newSelectedItems = unselectChild({
+      selectedItems,
+      parentRowIndex,
+      childRowIndex,
+      isTheOnlySelectedChildRow,
+    });
+
+    return newSelectedItems;
+  }
+
+  // SelectedParent but child is not selected.
+  const newSelectedItems = { ...selectedItems };
+
+  newSelectedItems[`${parentRowIndex}.${childRowIndex}`] = true;
+
+  return newSelectedItems;
+}
+
+function selectChild<T>({
+  childRowIndex,
+  parentRowIndex,
+  selectedItems,
+}: SelectRowArg<T>): TableSelectable['selectedItems'] {
+  const newSelectedItems = { ...selectedItems };
+
+  newSelectedItems[`${parentRowIndex}`] = true;
+  newSelectedItems[`${parentRowIndex}.${childRowIndex}`] = true;
+
+  return newSelectedItems;
+}
+
+function unselectChild<T>({
+  selectedItems,
+  parentRowIndex,
+  childRowIndex,
+  isTheOnlySelectedChildRow,
+}: SelectRowArg<T>) {
+  const newSelectedItems = { ...selectedItems };
+
+  delete newSelectedItems[`${parentRowIndex}.${childRowIndex}`];
+
+  if (isTheOnlySelectedChildRow) {
+    delete newSelectedItems[`${parentRowIndex}`];
+  }
+
+  return newSelectedItems;
```

Comment:
Again, we probably shouldn't mutate even if we are creating a new object/array. Maybe we can destructure where we delete instead?
```ts
const parentChildIndex = `${parentRowIndex}.${childRowIndex}`;
const { [parentChildIndex]: _child, ...removedChildSelectedItems } = selectedItems;

if (isTheOnlySelectedChildRow) {
  const parentIndex = `${parentRowIndex}`;
  const { [parentIndex]: _parent, ...removedParentSelectedItems } = removedChildSelectedItems;
   
  return removedParentSelectedItems;
}

return removedChildSelectedItems;
```